pyqt-apps/siriushla/common/afc_board/utils.py

```diff
@@ -0,0 +1,394 @@
+        'FruId':'FruId-Cte',
+        'BoardManuf':'BoardManuf-Cte',
+        'BoardName':'BoardName-Cte',
+        'BoardSN':'BoardSN-Cte',
+        'BoardPN':'BoardPN-Cte',
+        'ProdManuf':'ProdManuf-Cte',
+        'ProdName':'ProdName-Cte',
+        'ProdSN':'ProdSN-Cte',
+        'ProdPN':'ProdPN-Cte',
+        'PowerCtl':'PowerCtl-Sel',
+        'Pwr':'Pwr-Mon',
+        'SoftRst':'SoftRst-Cmd',
+        'SoftRstSts':'SoftRstSts-Mon'
+    }
+}
+
+BO_N = [1,3,6,8,11,13,16,18,21,23,26,28,31,33,36,38,41,43,46,48]
+BO_S = tuple([str(i) for i in BO_N])
+
+BO_N1 = [n+1 for n in BO_N]
+BO_S1 = tuple([str(i) for i in BO_N1])
+
+BO_N2 = [n+2 for n in BO_N]
+BO_S2 = tuple([str(i) for i in BO_N2])
+
+
+def device2slot(device):
+    device = SiriusPVName(device)
+    if device.sec == 'IA':
+        if device.dev == 'AMCFPGAEVR':
+            return 1
+
+        elif device.dev == 'FOFBCtrl':
+            return 2
+
+    elif device.sec == 'SI':
+        if device.sub[2:] in ['SBFE', 'SPFE', 'SAFE']:
+            return 4
+
+        elif device.sub[2:] == 'BCFE':
+            return 5
+
+        elif device.sub.endswith(('SA', 'SB', 'SP')):
+            return 6
+
+        elif device.sub.endswith(('M1', 'M2')):
+            return 7
+
+        elif device.sub.endswith(('C1')):
+            return 8
+
+        elif device.sub.endswith(('C2')) or device.sub.endswith(('C3')) and device.idx == '1':
+            return 9
+
+        elif device.sub.endswith(('C4')) or device.sub.endswith(('C3')) and device.idx == '2':
+            return 10
+
+    elif device.sec == 'TB':
+        if device.sub == '01':
+            return 6
+
+        elif device.sub == '02':
+            return 7
+
+        elif device.sub in ('03', '04'):
+            return 8
+
+    elif device.sec == 'TS':
+        if device.sub in ('01', '02'):
+            return 9
+
+        elif device.sub == '03' or device.sub == '04' and device.idx == '1':
+            return 10
+
+        elif device.sub == '04' and device.idx == '2':
+            return 11
+
+    elif device.sec == 'BO':
+        amc_bo = int(device.sub[:2])
+
+        if amc_bo in (BO_N + BO_N1):
+            return 11
+
+        elif amc_bo in BO_N2:
+            return 12
+
+
+def device2crate(device):
+    device = SiriusPVName(device)
+    if device.sec in ('TS', 'TB'):
+        return 'IA-20RaBPMTL'
+
+    elif device.sec == 'BO':
+        bo_num = int(device.sub[:2])
+
+        if bo_num in BO_N:
+            area_index = BO_N.index(bo_num) + 1
+
+        elif bo_num in BO_N1:
+            area_index = BO_N1.index(bo_num) + 1
+
+        elif bo_num in BO_N2:
+            area_index = BO_N2.index(bo_num) + 1
+
+        else:
+            return None
+
+        area_str = f"{area_index:02}"
+        return f'IA-{area_str}RaBPM'
+
+    elif device.sec == 'IA':
+        return f'IA-{device.sub}'
+
+    else:
+        return f'IA-{device.sub[:2]}RaBPM'
+
+
+class AFCAdvancedSettings(BaseWidget):
```

Comment:
I suggest moving this class to a separate module, once it is a graphical component (I try to keep in "utils" only auxiliary methods and constants, usually.